src/hacbs/pages/PipelineRunPage.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { Helmet } from 'react-helmet';
+import { PageSection, PageSectionVariants } from '@patternfly/react-core';
+import NamespacedPage from '../../components/NamespacedPage/NamespacedPage';
+import PageLayout from '../../components/PageLayout/PageLayout';
+import { useQuickstartCloseOnUnmount } from '../../hooks/useQuickstartCloseOnUnmount';
+import { getQueryArgument } from '../../shared/utils';
+import PipelineDetailsView from '../components/PipelineRunDetailsView/PipelineRunDetailsView';
+import PipelineRunsListView from '../components/PipelineRunListView/PipelineRunsListView';
+
+const PipelineRunPage = () => {
```

Comment:
You should be able to break this down into two separate components registered with two separate routes instead.

I know that you followed the way the ApplicationPage was written. But I think that page should also change. cc @rohitkrai03

Reply:
not sure, was following how appstudio was working. i don't think what you're saying is 100% accurate, you make it sound like they are one component, but list and details ARE two separate components, its page that loads one or the other based on if there's a name in the URL. but yes, the routes aren't uniquely defined routes in router, but they are unique. might want to delay this one for later.

Reply:
Oh i see. The pipelinerun name is in the `query` params :( 
That's very odd as the pattern we followed in openshift console was to use path segments for resource names. 

I'll connect with rohit to see why we have this new pattern.